tools/snp.sh

```diff
@@ -1112,138 +1100,136 @@ generate_snp_expected_measurement() {
+        sed 's|.*Measurement:\(.*\)Host Data.*|\1|g' | \
+        sed 's| ||g'")
 
-  # Check if 'snpguest' directory exists and remove it if it does
-  ssh_guest_command "
-  if [ -d \"snpguest\" ]; then
-     echo 'Directory snpguest exists. Removing it...'
-     rm -rf snpguest
-  else
-     echo 'Directory snpguest does not exist.'
-  fi
-"
+    # Remove any special characters and print the value
+    snpguest_report_measurement=$(echo "$snpguest_report_measurement" | sed 's/[^[:print:]\t]//g')
+    echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
 
-  # Clone the repository and build the project
-  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
-  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+    # Compare the expected measurement to the guest report measurement
+    if [ "${expected_measurement}" = "${snpguest_report_measurement}" ]; then
+      echo -e "The expected measurement matches the snp guest report measurement!"
+    else
+      >&2 echo -e "FAIL: measurements do not match"
+      return 1
+    fi
 }
 
-Verify_measurement(){
+setup_svsm_guest_attestation() {
+    ssh_guest_command <<'EOF'
+# Update and install necessary packages
+echo 'Updating package list...'
+sudo apt-get update
+
+echo 'Installing necessary packages...'
+sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+# Install Rust
+echo 'Installing Rust...'
+source "$HOME/.cargo/env" 2>/dev/null || true
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source "$HOME/.cargo/env" 2>/dev/null
+
+# Create the 'tools' directory if it does not exist
+if [ ! -d "tools" ]; then
+    echo 'Directory tools does not exist. Creating it...'
+    mkdir tools
+else
+    echo 'Directory tools already exists.'
+fi
+
+# Change to 'tools' directory
+cd tools || { echo 'Failed to change directory to tools'; exit 1; }
 
-  # Define the working directory and the output file
-  OUTPUT_FILE="$SETUP_WORKING_DIR/make_output.txt"
-  # Define the firmware file path
-  FW_FILE="$SETUP_WORKING_DIR/AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd"
+# Check if 'snpguest' directory exists and remove it if it does
+if [ -d "snpguest" ]; then
+    echo 'Directory snpguest exists. Removing it...'
+    rm -rf snpguest
+else
+    echo 'Directory snpguest does not exist.'
+fi
 
-  cd "$SETUP_WORKING_DIR"/AMDSEV/svsm
+# Clone the repository into the 'tools' directory
+git clone https://github.com/virtee/snpguest.git
 
-  # Run the make command and save the output to a file
-  FW_FILE="$FW_FILE" make &> "$OUTPUT_FILE"
+# Build the project inside the 'snpguest' directory
+cd snpguest || { echo 'Failed to change directory to snpguest'; exit 1; }
+source "$HOME/.cargo/env"
+cargo build --release
 
-  # Initialize the variable for Launch Digest
-  launch_digest=""
+# Copy the built binary to the home directory
+cp target/release/snpguest "$HOME/"
+EOF
+}
```

Comment:
This entire function can be removed. The program is already building the tool once. It doesn't make sense to duplicate this again.